tornado/test/tcpserver_test.py

```diff
@@ -126,11 +126,11 @@ def run_subproc(self, code):
         proc = subprocess.Popen(
             sys.executable, stdin=subprocess.PIPE, stdout=subprocess.PIPE
         )
-        proc.stdin.write(utf8(code))
-        proc.stdin.close()
+        proc.stdin.write(utf8(code))  # type: ignore[union-attr]
```

Comment:
If this is the error I think it is, my preferred solution is an `assert proc.stdin is not None` after the call to subprocess.Popen instead of a type-ignore comment on each reference.

Reply:
I had another look at this function and it turns out this could be replaced with `subprocess.run` since python3.5 support was dropped